### PR TITLE
Themes API

### PR DESCRIPTION
**Pull request by Conner1115**
Added the themes API, including a `useThemeValues` hook.  I removed the old `useTheme` hook and the older types associated with it since it was undocumented and the types were all wrong and dreadfully messy.

### CODE REVIEW

File: src/types/index.ts [hunk 1]
```diff
@@ -49,124 +49,110 @@ export interface DirectoryChildNode {
...
 }
 
-export interface Theme {
-  id: string;
-  description: string;
-  values: ThemeValues;
-  colorScheme: ColorScheme;
-  name: string;
-  isOfficial: boolean;
-}
```
Comment (lunaroyster) [inline src/types/index.ts:168]: i feel like these are still useful values for certain extensions to use: for example, theme managers. I don't see the value in removing them. 

having second thoughts about this migration. It's not like we're simplifying the number of tokens we're passing to the 6 base tokens

Comment (lunaroyster) [reply src/types/index.ts:168]: happy to approve an addition only version of this change. let's preserve useTheme for now

File: src/react/useTheme.ts
Comment (lunaroyster) [inline src/react/useTheme.ts:1]: or this

**Comment by lunaroyster** [comment]
let's preserve the old theme types (and make them more correct). We can just add the new theme values methods

**Review by lunaroyster — APPROVED** [review]

File: src/types/index.ts [hunk 2]
```diff
@@ -49,122 +49,164 @@ export interface DirectoryChildNode {
...
-    brownStronger: CssColor;
-    brownStrongest: CssColor;
-    black: CssColor;
-    white: CssColor;
-  };
-  editor: {
-    //   syntaxHighlighting: Array<SyntaxHighlightingSpec>;
-    syntaxHighlighting: any;
-  };
+export interface ThemeValuesGlobal {
```
[the remark this answers is not in the document]
Comment (Conner1115) [reply src/types/index.ts:52]: I'll merge this first, and when we can figure out how to make typedoc work, we can make a separate PR.

**Pull request merged**